flash/data/data_pipeline.py

```diff
@@ -0,0 +1,612 @@
+                                              uncollate
+                                                  │
+                                          per_sample_transform
+
+    """
+
+    PREPROCESS_FUNCS = {
+        "load_data", "load_sample", "per_sample_pre_tensor_transform", "per_sample_to_tensor_transform",
+        "per_sample_post_tensor_transform", "per_batch_transform", "per_sample_transform_on_device",
+        "per_batch_transform_on_device", "collate"
+    }
+    # TODO: unused?
+    POSTPROCESS_FUNCS = ("per_batch_transform", "per_sample_transform", "save_data", "save_sample")
+
+    def __init__(self, preprocess: Optional[Preprocess] = None, postprocess: Optional[Postprocess] = None) -> None:
+        if preprocess is None:
+            preprocess = Preprocess()
+
+        if postprocess is None:
+            postprocess = Postprocess()
+
+        self._preprocess_pipeline = preprocess
+        self._postprocess_pipeline = postprocess
+        self._postprocessor = None
+        self._running_stage = None
+
+    @staticmethod
+    def _is_overriden(method_name: str, process_obj, super_obj: Any, prefix: Optional[str] = None) -> bool:
+        """
+        Cropped Version of
+        https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/model_helpers.py
+        """
+
+        current_method_name = method_name if prefix is None else f'{prefix}_{method_name}'
+
+        if not hasattr(process_obj, current_method_name):
+            return False
+
+        return getattr(process_obj, current_method_name).__code__ != getattr(super_obj, method_name).__code__
+
+    @classmethod
+    def _is_overriden_recursive(
+        cls, method_name: str, process_obj, super_obj: Any, prefix: Optional[str] = None
+    ) -> bool:
+        """
+        Cropped Version of
+        https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/model_helpers.py
+        """
+        if prefix is None and not hasattr(super_obj, method_name):
+            raise MisconfigurationException(f"This function doesn't belong to the parent class {super_obj}")
+
+        current_method_name = method_name if prefix is None else f'{prefix}_{method_name}'
+
+        if not hasattr(process_obj, current_method_name):
+            return False or DataPipeline._is_overriden_recursive(method_name, process_obj, super_obj)
+
+        has_different_code = getattr(process_obj,
+                                     current_method_name).__code__ != getattr(super_obj, method_name).__code__
+        if prefix is None:
+            return has_different_code
+        else:
+            return has_different_code or cls._is_overriden_recursive(method_name, process_obj, super_obj)
+
+    @staticmethod
+    def _do_nothing_collate(samples: Sequence[Any]) -> Sequence[Any]:
+        return samples
+
+    @staticmethod
+    def _do_nothing_uncollate(batch: Any) -> Any:
+        return batch
```

Comment:
in fact, do you need both, would be enough to define just one
```python
    @staticmethod
    def _void(data: Any) -> Any:
        return data
```

Reply:
renamed _identity